From 71cced479ef273427405d644ab3047909a93d9a2 Mon Sep 17 00:00:00 2001
From: Tobias Kortkamp <t@tobik.me>
Date: Fri, 9 Apr 2021 06:58:31 +0200
Subject: [PATCH] Attempt to fix build order bug

Uncovered while building with samurai [0] but can probably also
happen with ninja eventually.

In file included from ../src/mate-applet/applet.c:19:
In file included from ../src/frontend/classic/classic-window.h:14:
../src/frontend/classic/../menu-private.h:18:10: fatal error: 'libsaver-glue.h' file not found
 #include "libsaver-glue.h"
         ^~~~~~~~~~~~~~~~~
1 error generated.
samu: subcommands failed

[0] https://github.com/michaelforney/samurai
---
 src/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/meson.build b/src/meson.build
index 7d8ca3f..c6750f1 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -13,6 +13,7 @@ libresources = static_library(
 
 link_libresources = declare_dependency(
     link_with: libresources,
+    sources: brisk_resources,
     include_directories: [
         data_include,
     ],
@@ -38,6 +39,7 @@ libsession_stub = static_library(
 # Allow linking to the stub
 link_libsession_stub = declare_dependency(
     link_with: libsession_stub,
+    sources: [libsaver_glue, libsession_glue],
     dependencies: [
         dep_gio_unix,
     ],
